src/source/CppGenerator.scala

```diff
@@ -154,6 +153,11 @@ class CppGenerator(spec: Spec) extends Generator(spec) {
     def writeCppPrototype(w: IndentWriter) {
       writeDoc(w, doc)
       writeCppTypeParams(w, params)
+      if (r.ext.cpp) {
```

Comment:
You're inserting this between the struct declaration and its template parameters an documentation which are above.  I think you want this to be at the top of the writeCppPrototype function.

Alternately, I think you could just add an element to refs.hppFwds, though you'd have to format it so it works outside of the namespace: "namespace foo { struct bar; };"